vita3k/modules/include/modules/tracy.h

```diff
@@ -28,6 +29,14 @@ std::string to_debug_str(const MemState &mem, T data) {
     return datass.str();
 }
 
+// Override for guest pointers, we want to print the guest address
+template <typename U>
+std::string to_debug_str(const MemState &mem, Ptr<U> data) {
+    std::stringstream datass;
+    datass << data.address();
+    return datass.str();
+}
+
```

Comment:
also this (or something like) can be added to log addresses of pointers
`template <typename U>
std::string to_debug_str(const MemState &mem, U* data) {
    return log_hex(Ptr<U>(data, mem).address());
}`

Reply:
hmm should i use log_hex or log_hex_full? just saying because tracy will execute this pointer template on most exports because most of them have pointers in them, log_hex uses fmt::format and log_hex_full uses stringstreams, im saying this because of performance, idk if it even really matters :thinking:

Reply:
did a test and log_hex_full is double the efficient as log_hex, taking always half the time than log_hex, ig i will use log_hex_full

Reply:
added log_hex_fast, which is the same implementation as log_hex_full but without the padding, so in theory it should be on par or a tiny bit faster and should have same output as log_hex, i added a TODO to replace the fmt one to this faster one in case its needed, this is how it gets displayed on tracy:
![image](https://user-images.githubusercontent.com/30161277/194680941-d1f6ec87-8690-4985-9788-41dc64a5be68.png)

Reply:
Thank you for checking performance, I did not know about log_hex_full
userAgent value on screenshot looks too long for 32 bit value, it seems wrong to me, maybe the code has an error, i'll check.
I forgot about char* It's needed a separate template specification for this. Something like 
template <> std::string to_debug_str(const MemState &mem, char* data) { return std::string(data ? data : "0x0 NULL"); }

Reply:
I see now. Pointers in Sce functions are usually guest pointers converted to host pointers. So for log I convert them back to guest pointers (Ptr<U>(data, mem).address()). I don't know what version is more usable (host or guest) but someone asks for guest pointer values

Reply:
Added a template for char*, printing both the address and the contents, For pointers i decided its more consistent to print the host address instead as its more easy to handle imo, unless there is some easy way to convert pointers back to guest i guess its just better to choose host in the meantime

Reply:
way to convert pointers back to guest is Ptr(host_pointer, mem).address()
But i think it's good enough now.